src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -250,6 +268,109 @@ private EmailWrapper generateSubmissionConfirmationEmail(
 
     }
 
+    private EmailWrapper generateSessionLinksRecoveryNonExistingEmailForStudent(String recoveryEmailAddress) {
+        String emailBody;
+        String subject = EmailType.SESSION_LINKS_RECOVERY.getSubject();
+        String recoveryUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSIONS_LINK_RECOVERY_PAGE).toAbsoluteString();
+        emailBody = Templates.populateTemplate(
+                EmailTemplates.SESSION_LINKS_RECOVERY_EMAIL_NOT_FOUND,
+                "${userEmail}", SanitizationHelper.sanitizeForHtml(recoveryEmailAddress),
+                "${supportEmail}", Config.SUPPORT_EMAIL,
+                "${teammateHomePageLink}", Config.APP_URL,
+                "${sessionsRecoveryLink}", recoveryUrl);
+        EmailWrapper email = getEmptyEmailAddressedToEmail(recoveryEmailAddress);
+        email.setSubject(subject);
+        email.setContent(emailBody);
+        return email;
+    }
+
+    private EmailWrapper generateSessionLinksRecoveryEmailForStudent(String recoveryEmailAddress) {
+        String emailBody;
+        String subject = EmailType.SESSION_LINKS_RECOVERY.getSubject();
+
+        Instant endTime = Instant.now();
+        Instant startTime = endTime.minus(Duration.ofDays(180));
+        Map<String, StringBuilder> linkFragmentsMap = new HashMap<>();
+        String studentName = null;
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllFeedbackSessionsWithinTimeRange(startTime, endTime);
+
+        for (FeedbackSessionAttributes session : sessions) {
+            String courseId = session.getCourseId();
+            CourseAttributes course = coursesLogic.getCourse(courseId);
+            StudentAttributes student = studentsLogic.getStudentForEmail(courseId, recoveryEmailAddress);
+            StringBuilder linksFragmentValue;
+            if (linkFragmentsMap.containsKey(courseId)) {
+                linksFragmentValue = linkFragmentsMap.get(courseId);
+            } else {
+                linksFragmentValue = new StringBuilder(5000);
+            }
+
+            if (student != null) {
+                studentName = student.getName();
+                String submitUrlHtml = "";
+                String reportUrlHtml = "";
+
+                if (session.isOpened() || session.isClosed()) {
+                    String submitUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSION_SUBMISSION_PAGE)
```

Comment:
this link as well. It shld be something like `http://localhost:4200/web/student/sessions/submission?...`, but now your result is `http://localhost:4200/web/sessions/submission?...`

Reply:
Yes, I thought the page haven't been migrated yet. I have fixed it after pull from latest master